src/library/text_writer.h

```diff
@@ -290,6 +290,102 @@ namespace xlang::text
         std::vector<char> m_first;
     };
 
+
+    template <typename T>
+    struct indented_writer_base : public writer_base<T>
+    {
+        struct indent_guard
+        {
+            explicit indent_guard(indented_writer_base<T>& w, int32_t offset = 1) noexcept : _writer(w), _offset(offset)
+            {
+                _writer.indent += _offset;
+            }
+
+            ~indent_guard()
+            {
+                _writer.indent -= _offset;
+            }
+
+        private:
+            indented_writer_base<T>& _writer;
+            int32_t _offset;
```

Comment:
Initialize just for the sake of paranoia.

Reply:
Sure :)